### PR TITLE
Resign first responder when hiding keyboard

### LINKED ISSUE #610: Resign first responder on iPad when dismissing kb (when using an external keyboard)
<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.11.0 (2101)
Device: iPhone 7 Plus (iOS 11.0.3)
TestFlight: true
</pre>
</details>

### PR DESCRIPTION
**Pull request by BasThomas**
Fixes #610 

Ended up doing this when the keyboard will hide, so it happens at that point immediately. Thought that would make more sense then doing so when the view will disappear.

### CODE REVIEW

File: Classes/Search/SearchViewController.swift [hunk 1]
```diff
@@ -44,6 +44,7 @@ SearchResultSectionControllerDelegate {
     init(client: GithubClient) {
         self.client = client
         super.init(nibName: nil, bundle: nil)
+         NotificationCenter.default.addObserver(self, selector: #selector(SearchViewController.willHideKeyboard), name: .UIKeyboardWillHide, object: nil)
```
Comment (rnystrom) [inline Classes/Search/SearchViewController.swift]: nit: prefer params on their own line, makes it more readable

**Comment by BasThomas** [comment]
@rnystrom how would I do that target-action with a notificationcenter observer?

**Comment by rnystrom** [comment]
It’s just an objc selector, so you should be able to send a param-less method like `resignFirstResponder`. So `searchBar` as the target and `UISearchBar.resignFirstResponder` as the action? Something like that.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BasThomas** [comment]
Trying to do that, I can't seem to get around the `Argument of '#selector' does not refer to an '@objc' method, property, or initializer`, whatever I do.

**Comment by BasThomas** [comment]
Ah, cool. Fixed.

**Comment by rnystrom** [comment]
Super clean, love it.

**Comment by BasThomas** [comment]
Thanks for the help as well! :D

**Pull request merged**